src/providers/oracle/qgsoracleconn.cpp

```diff
@@ -37,7 +37,9 @@ QMap<QString, QDateTime> QgsOracleConn::sBrokenConnections;
 QgsOracleConn *QgsOracleConn::connectDb( const QgsDataSourceUri &uri, bool transaction )
 {
   const QString conninfo = toPoolName( uri );
-  if ( !transaction )
+  const bool shared = !transaction && QApplication::instance()->thread() == QThread::currentThread();
```

Comment:
The sql server provider uses a different approach here -- see https://github.com/qgis/QGIS/blob/master/src/providers/mssql/qgsmssqldatabase.cpp#L51 . This allows use of shared connections from the same non-main thread too. Could that approach be utilised here?

Reply:
Yes, but not by patching connection name with thread id because it would not work nicely with QgsConnectionPool in oracle (it's not used mssql). 

But I made the [modifications](https://github.com/qgis/QGIS/pull/53201/commits/80b08ddbe67114e22092977748a7409777c66950) to make it happen